File: javascript/easy/2417.sortPeople.js
/**
 * @param {string[]} names
 * @param {number[]} heights
 * @return {string[]}
 */
var sortPeople = function (names, heights) {
  const people = names.map((name, index) => ({
    name: name,
    height: heights[index],
  }));

  people.sort((a, b) => b.height - a.height);

  return people.map((person) => person.name);
};

console.log(sortPeople(["Mary", "John", "Emma"], [180, 165, 170])); // ["Mary","Emma","John"]
// Explanation: Mary is the tallest, followed by Emma and John.

console.log(sortPeople(["Alice", "Bob", "Bob"], [155, 185, 150])); // ["Bob","Alice","Bob"]
// Explanation: The first Bob is the tallest, followed by Alice and the second Bob.

File: python/easy/2417.sortPeople.py
from typing import List


class Solution:
    def sortPeople(self, names: List[str], heights: List[int]) -> List[str]:
        people = [{"name": name, "height": heights[i]} for i, name in enumerate(names)]

        people.sort(key=lambda person: person["height"], reverse=True)

        return [person["name"] for person in people]


if __name__ == "__main__":
    solution = Solution()
    print(
        solution.sortPeople(["Mary", "John", "Emma"], [180, 165, 170])
    )  # ["Mary","Emma","John"]
    print(
        solution.sortPeople(["Alice", "Bob", "Bob"], [155, 185, 150])
    )  # ["Bob","Alice","Bob"]
